parquet-common/src/main/java/org/apache/parquet/bytes/SingleBufferInputStream.java

```diff
@@ -88,6 +132,15 @@ public long skip(long n) {
     return bytesToSkip;
   }
 
+  @Override
+  public void skipFully(long n) throws IOException {
+    try {
+      buffer.position(buffer.position() + (int)n);
```

Comment:
Looks like this is just trying to avoid the checks that are being done in `skip`. I don't think that's a good idea. This should delegate to skip instead.

Reply:
I did this specifically because it's performance-critical. I did a bunch of profiling, and skips are among the operations that have to have minimal overhead. Delegating to skip() would introduce a bunch of checks that the JIT isn't going to be smart enough to remove.